kie-internal/src/main/java/org/kie/internal/task/api/prediction/PredictionService.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.internal.task.api.prediction;
+
+import org.kie.api.task.model.Task;
+
+import java.util.Map;
+
+/**
+ * Interface which allows a prediction or recommendation service implementation to be trained with task data
+ * and return outcome predictions based on task input.
+ *
+ * This interface is still considered subject to change.
```

Comment:
@ruivieira so this can still change in future versions? Are we ok with that?

Reply:
@MarianMacik yes this is fine